tests/test_ptable.py

```diff
@@ -49,7 +45,7 @@ class TestDataPreprocessor:
         "Li": np.array([6.0, 8.0]),  # float array
         "Na": 11.0,  # float
         "Mg": {"a": -1, "b": 14.0}.values(),  # dict_values
-        "Al": {-1, 2.3},  # mixed int/float set
```

Comment:
I don't think user should use unordered data for plotting, this is not supported by `pd.DataFrame` constructor:
```python
Traceback (most recent call last):
  File "/Users/yang/developer/pymatviz/test_df.preprocess.py", line 18, in <module>
    input_df_0: pd.DataFrame = pd.DataFrame(test_dict)
                               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/frame.py", line 778, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 119, in arrays_to_mgr
    arrays, refs = _homogenize(arrays, index, dtype)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 629, in _homogenize
    val = sanitize_array(val, index, dtype=dtype, copy=False)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/construction.py", line 642, in sanitize_array
    _sanitize_non_ordered(data)
  File "/Users/yang/developer/pymatviz/venv/lib/python3.12/site-packages/pandas/core/construction.py", line 693, in _sanitize_non_ordered
    raise TypeError(f"'{type(data).__name__}' type is unordered")
TypeError: 'set' type is unordered
```

Reply:
unordered data is fine for a histogram, no? i think in this case, `DataFrame` has stricter requirements than we need to enforce

Reply:
Yes it's fine for histogram, but this `TestDataPreprocessor` is for more than histogram.

Reply:
Maybe I wasn't make myself clear, the problem is not about our data processor, pandas cannot construct a `df` from a dict with `set` as values:
```python
test_dict = {
        "H": 1,  # int
        "He": {2.0, 4.0},  # float list
    }

input_df_0: pd.DataFrame = pd.DataFrame(test_dict)  # TypeError: 'set' type is unordered
```

Reply:
No that part I understood :)